google/resource_pubsub_topic_iam_test.go

```diff
@@ -0,0 +1,215 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"reflect"
+	"sort"
+	"testing"
+)
+
+func TestAccPubsubTopicIamBinding(t *testing.T) {
+	t.Parallel()
+
+	topic := "test-topic-iam-" + acctest.RandString(10)
+	account := "test-topic-iam-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// Test IAM Binding creation
+				Config: testAccPubsubTopicIamBinding_basic(topic, account),
+				Check: testAccCheckPubsubTopicIam(topic, "roles/pubsub.publisher", []string{
+					fmt.Sprintf("serviceAccount:%s-1@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+				}),
+			},
+			{
+				// Test IAM Binding update
+				Config: testAccPubsubTopicIamBinding_update(topic, account),
+				Check: testAccCheckPubsubTopicIam(topic, "roles/pubsub.publisher", []string{
+					fmt.Sprintf("serviceAccount:%s-1@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+					fmt.Sprintf("serviceAccount:%s-2@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+				}),
+			},
+			{
+				ResourceName:  "google_pubsub_topic_iam_binding.foo",
+				ImportStateId: fmt.Sprintf("%s roles/pubsub.publisher", getComputedTopicName(getTestProjectFromEnv(), topic)),
+				ImportState:   true,
+			},
+		},
+	})
+}
+
+func TestAccPubsubTopicIamMember(t *testing.T) {
+	t.Parallel()
+
+	topic := "test-topic-iam-" + acctest.RandString(10)
+	account := "test-topic-iam-" + acctest.RandString(10)
+	accountEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// Test Iam Member creation (no update for member, no need to test)
+				Config: testAccPubsubTopicIamMember_basic(topic, account),
+				Check: testAccCheckPubsubTopicIam(topic, "roles/pubsub.publisher", []string{
+					fmt.Sprintf("serviceAccount:%s", accountEmail),
+				}),
+			},
+			{
+				ResourceName:  "google_pubsub_topic_iam_member.foo",
+				ImportStateId: fmt.Sprintf("%s roles/pubsub.publisher serviceAccount:%s", getComputedTopicName(getTestProjectFromEnv(), topic), accountEmail),
+				ImportState:   true,
+			},
+		},
+	})
+}
+
+func TestAccPubsubTopicIamPolicy(t *testing.T) {
+	t.Parallel()
+
+	topic := "test-topic-iam-" + acctest.RandString(10)
+	account := "test-topic-iam-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// Test Iam Member creation (no update for member, no need to test)
```

Comment:
This comment is misplaced.
Also, should we have an update test for this?

Reply:
Done